src/MissionManager/LandingComplexItem.cc

```diff
@@ -388,13 +407,15 @@ bool LandingComplexItem::_scanForItem(QmlObjectListModel* visualItems, bool flyV
     MissionItem& missionItemFinalApproach = item->missionItem();
     if (missionItemFinalApproach.command() == MAV_CMD_NAV_LOITER_TO_ALT) {
         if (missionItemFinalApproach.frame() != landPointFrame ||
-                missionItemFinalApproach.param1() != 1.0 || missionItemFinalApproach.param3() != 0 || missionItemFinalApproach.param4() != 1.0) {
+            (masterController->managerVehicle()->firmwareType() == MAV_AUTOPILOT_ARDUPILOTMEGA ?
```

Comment:
Should use `masterController->managerVehicle()->apmFirmware`. Same in other places